scroll-animations-1/Overview.bs

```diff
@@ -353,20 +354,113 @@ spec: cssom-view-1; type: dfn;
 
+	The [=used value=] of 'scroll-timeline-axis'
+	is the axis of the [=current attachment=],
+	or the initial value if there is no [=current attachment=].
+
+### Attachment of a Scroll Progress Timeline: the 'scroll-timeline-attachment' property ### {#scroll-timeline-attachment}
+
+	<pre class='propdef'>
+	Name: scroll-timeline-attachment
+	Value: [ local | defer | ancestor ]#
+	Initial: local
+	Applies to: all elements
+	Inherited: no
+	Computed value: a list of the keywords specified
+	Animation type: not animatable
+	</pre>
+
+	Specifies attachment behavior for each [=scroll progress timeline=]
+	associated with this element.
+
+	A scroll timeline has a set of <dfn>attachments</dfn>,
+	that determine which [=scroll container=] and {{ScrollTimeline/axis}} to use
+	when calculating the [progress of the timeline](#scroll-timeline-progress).
+
+	<dl>
+		<dt><dfn>'local'</dfn>
+		<dd>
+			The [=attachments=] of the timeline is a single item set
+			containing that timeline.
+
+		<dt><dfn>'defer'</dfn>
+		<dd>
+			The [=attachments=] is the set of timelines in the flat-tree descendants
+			that satisfy all of the following:
+			 - The timeline has a matching 'scroll-timeline-name'.
+			 - The timeline has [=ancestor=] attachment.
+			 - The timeline has no intermediate [=defer|deferred=] timeline
+			   that would capture it.
+
+		<dt><dfn>'ancestor'</dfn>
+		<dd>
+			The set of [=attachments=] is empty.
+
+			Timelines with [=ancestor=] attachment are treated as nameless
+			for the purposes of timeline lookup;
+			they can't be referenced by 'animation-timeline'.
+	</dl>
+
+	The <dfn>current attachment</dfn> of a timeline is determined as follows:
+
+	<dl class="switch">
+		:   If the set of [=attachments=] contains exactly one timeline
+		::  The [=current attachment=] is that timeline.
+
+		:   Otherwise,
+		::  There is no [=current attachment=].
+	</dl>
+
+	<div class=example>
+		Using a [=defer|deferred timeline=],
+		an element can effectively refer to timelines
+		that appear <em>after it</em> in tree order.
+		For example, the following creates an animation on <code>#target</code> that
+		is ultimately linked to a scroll timeline defined on the subsequent sibling.
+
+		<pre class=lang-css>
+		@keyframes anim {
+		  from { color: red; }
+		  to { color: green; }
+		}
+
+		#root {
+		  scroll-timeline-name: mytimeline;
+		  scroll-timeline-attachment: defer;
+		}
+
+		#root #target {
+			animation: anim auto;
+			animation-timeline: mytimeline;
+		}
+
+		#root #target + #scroller {
+			scroll-timeline-name: mytimeline;
+		  scroll-timeline-attachment: ancestor;
+		}
+		</pre>
+	</div>
+
 ### Scroll Timeline Shorthand: the 'scroll-timeline' shorthand ### {#scroll-timeline-shorthand}
 
 	<pre class='propdef shorthand'>
 	Name: scroll-timeline
-	Value: [ <<'scroll-timeline-name'>> <<'scroll-timeline-axis'>>? ]#
+	Value: [ <<'scroll-timeline-name'>> [ <<'scroll-timeline-axis'>> | <<'scroll-timeline-attachment'>> ] ? ]#
```

Comment:
This makes it impossible for the descendent to declare the axis unfortunately. In order to avoid an axis specified on the deferred attachment It would have to be split up into that value which I'm not sure is something we do. e.g.

```
Value: [ <<'scroll-timline-name'>> 'defer' | <<'scroll-timline-name'>> <<'scroll-timeline-axis'>>? 'ancestor'? ]#
```

I'm fine with just doing the simple thing and having them all in the shorthand if this isn't speccable.

Reply:
Ah, yes, good point. Fixed (did the simple thing for now).

Although something like what you said does seem speccable. "Splitting" the value to achieve something special doesn't seem too far off from what we're doing in e.g. https://drafts.csswg.org/css-text-4/#white-space-property.

So maybe we could do:  `<<'scroll-timeline-name'>> [ defer | [ <<'scroll-timeline-axis'>> ancestor? ] ]?`

But maybe we leave this for a separate PR, for maximum bikeshedding.